### ISSUE TITLE
Reversing progress start animation from 0.0%

### ISSUE THREAD

**Issue by joelbrostrom** [diegoveloper/flutter_percent_indicator#15]
When you increase the percentage with `animateFromLastPercent = true` the animation starts from the last value and ends at the new value.
However when reducing the value the animation starts from 0 and ends at the new value.

For example:
50% -> 75%  animates from  50% to 75%
50% -> 25% animates from 0% to 25%

Is there support for reversing  animation if `newValue < currentValue`?

**Comment by joelbrostrom** [comment]
Is this plugin no longer supported?

**Comment by diegoveloper** [comment]
Yes, it's still supported. Could you add a minimum sample code to reproduce the issue please?

**Comment by joelbrostrom** [comment]
Something like:

```class PagerState extends State<MyPager> {
  PageController _pageController = PageController();
  List<PageData> pages;
  double _progress;

  @override
  Widget build(BuildContext context) {
    return Column(children: [
      PageView.builder(
          key: PageStorageKey('Questionnarie'),
          controller: _pageController,
          itemCount: pages.length,
          itemBuilder: (context, index) {
            return SomePage(pages[index]);
          },
          onPageChanged: (index) {
            setState(() {
              _progress = index / pages.length;
            });
          }),
      LinearPercentIndicator(
        alignment: MainAxisAlignment.center,
        width: MediaQuery.of(context).size.width - 50,
        animation: true,
        lineHeight: 15.0,
        animationDuration: 1000,
        percent: _progress,
        animateFromLastPercent: true,
        center: Text("${(100 * _progress).round()}%"),
        linearStrokeCap: LinearStrokeCap.roundAll,
        progressColor: Colors.greenAccent,
      ),
    ]);
  }
}
```
When scrolling nextPage the animation continues from the previous value, but when scrolling to previous page it starts from 0 and animates to the new value.

**Comment by joelbrostrom** [comment]
Here, I made a better example app to illustrate the issue.
Just copy paste and add `percent_indicator: ^1.0.13` to pubspec :)

```
import 'package:flutter/material.dart';
import 'package:percent_indicator/linear_percent_indicator.dart';

void main() =>  runApp(App());

class App extends StatelessWidget {
  @override
  Widget build(BuildContext context) => MaterialApp(home: MyProgressBar() );
}

class MyProgressBar extends StatefulWidget {
  @override
  State<StatefulWidget> createState() => MyProgressBarState();
}

class MyProgressBarState extends State<MyProgressBar> {
  double _progress;

  @override
  void initState() {
    super.initState();
    _progress = 0.0;
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
        body: Center(
            child: LinearPercentIndicator(
          width: MediaQuery.of(context).size.width,
          animation: true,
          animateFromLastPercent: true,
          percent: _progress,
        )),
        floatingActionButton: Row(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            FloatingActionButton(
                child: Icon(Icons.remove),
                onPressed: () {
                  setState(() {
                    _progress -= 0.1;
                  });
                }),
            FloatingActionButton(
                child: Icon(Icons.add),
                onPressed: () {
                  setState(() {
                    _progress += 0.1;
                  });
                })
          ],
        ));
  }
}

```

**Comment by diegoveloper** [comment]
Thanks, I'll check this issue later today

**Comment by joelbrostrom** [comment]
Sure, If you wan't I can take a look at it and if I figure it out I can send a PR.
Do you accept contributions?

**Comment by diegoveloper** [comment]
Of course, everyone who uses this package would appreciate it. Can you take
a look the issue?

**Comment by joelbrostrom** [comment]
I'm two beers in and after work party starts in 1h 30m but I'll se what I can do :)

**Comment by diegoveloper** [comment]
hahaha, have fun bro :)

**Comment by joelbrostrom** [comment]
I opened a PR with the fix for this issue.
It was a <sup><sub>tiny</sub></sup> mistake, so there is not a lot to review.
Great lib.
Ty for the communication :)

**Comment by diegoveloper** [comment]
awesome! thanks, it was merged

**Comment by joelbrostrom** [comment]
Can you push a new version (1.0.14) with this change?
I need it in my project :)

**Comment by diegoveloper** [comment]
Sure, but first, could you check the same issue for circular percent
indicator?

**Comment by joelbrostrom** [comment]
Done, sorry for the delay.

**Comment by diegoveloper** [comment]
Thanks bro, it was published : 1.0.14
